Initial draft of security reporting processs (#1431)
o3de/o3de.org: content/docs/contributing/security/_index.md

---
linktitle: Security
title: "Contributing to O3DE: Security"
toc: true
weight: 100
---

## Open 3D Engine: Security

> This page describes O3DE's Security and Disclosure Policies. 

Reviewer: ```suggestion
This page describes the Security and Disclosure policies for **Open 3D Engine (O3DE)**. 
```

- Style guide application: No possessive on project name, formatting

Author: Fixed

O3DE values all contributions that help keep O3DE secure. [SIG-Security](https://github.com/o3de/sig-security) volunteers will thoroughly research and investigate all reported vulnerabilities. 

> # Report a Vulnerability

Reviewer: ```suggestion
# How to report a vulnerability

```

Author: Fixed

> We encourage all vulnerabilities to be reported through the security@o3de.org list. Vulnerability reports should follow the same information reporting format as [standard O3DE issues](https://github.com/o3de/o3de/blob/development/.github/ISSUE_TEMPLATE/bug_template.md) along with any security specific details.

Reviewer 2: security@o3de.org `mailing` list.

Author: Fixed.

> We encourage all vulnerabilities to be reported through the security@o3de.org mailing list. Vulnerability reports should follow the same information reporting format as [standard O3DE issues](https://github.com/o3de/o3de/blob/development/.github/ISSUE_TEMPLATE/bug_template.md) along with any security specific details.

Reviewer: ```suggestion
We encourage all vulnerabilities to be reported through the [security@o3de.org](mailto:security@o3de.org) mailing list. Vulnerability reports should follow the same information reporting format as [standard O3DE issues](https://github.com/o3de/o3de/blob/development/.github/ISSUE_TEMPLATE/bug_template.md) along with any security specific details.
```

Add `mailto:` link to email address.

Author: Fixed

> ## When Should I Report a Vulnerability?

Reviewer: ```suggestion
## When should I report a vulnerability?

```

Author: Fixed

* You have discovered a potential security issue relating to O3DE. 
* You have discovered or are aware of a security issue relating to an O3DE dependency that may impact O3DE.

>      * If the issues is in a product that O3DE depends upon, we strongly encourage you to engage with that project's vulnerability reporting mechanisms. 

Reviewer 3: should we say "...encourage you to **also** engage with that project's vulnerability reporting mechanisms"? (I'm assuming we'd still want the report)

Author: Yes. Good catch. Fixed.

> ## When Should I NOT Report a Vulnerability?

Reviewer: ```suggestion
## When should I NOT report a vulnerability?

```

Author: Fixed

> * You need help setting up or have issues relating to the use of security dependencies such as OpenSSL or TLS/DTLS 

Reviewer: ```suggestion
* You need help setting up or have issues relating to the use of security dependencies such as OpenSSL or TLS/DTLS.
```

* You need help applying a security related fix to your O3DE applications. 
* The issue is not related to security. Please report issue using the normal [O3DE issue reporting process](https://github.com/o3de/o3de/issues).

> ## Can I Report a Vulnerability via GitHub Issues?

Reviewer: ```suggestion
## Can I report a vulnerability via GitHub Issues?

```

> If there is public disclosed security information, such as a report in the [National Vulnerability Database](https://nvd.nist.gov/) then GitHub issues can be used to communicate security vulnerabilities impacting O3DE.

Reviewer 2: `publicly` disclosed security information

Author: Fixed.

> # Security Vulnerability Response

Reviewer: ```suggestion
# Security vulnerability response

```

Author: Fixed

> When a security vulnerability is reported via security@o3de.org, a member of the SIG-Security response team should respond within 3 working days. Once the issue has been acknowledged, the SIG-Security response team will follow its [documented response process](https://github.com/o3de/sig-security) to ensure the reporter is updated and informed of current actions and next steps.

Reviewer 3: I feel like we should make the "**documented response process**" link go somewhere real before publishing this (even the in-flight response process RFC), unless we plan to add a very visible "RESPONSE PROCESS" link to the `sig-security` README/landing page in the near future

Author: Theres nothing good to link to so am erring on the side that we will fix this up immediately on publication of the RESPONSE_PROCESS.md or similar (once the RFC is accepted - I could link to the RFC I suppose). For now response is adhoc and owned by SIG-Security.

> When a security vulnerability is reported via security@o3de.org, a member of the SIG-Security response team should respond within 3 working days. Once the issue has been acknowledged, the SIG-Security response team will follow its [documented response process](https://github.com/o3de/sig-security/issues/14) to ensure the reporter is updated and informed of current actions and next steps.

Reviewer: ```suggestion
When a security vulnerability is reported via the [security mailing list](mailto:security@o3de.org), a member of the SIG-Security response team should respond within 3 working days. Once the issue has been acknowledged, the SIG-Security response team will follow its [documented response process](https://github.com/o3de/sig-security/issues/14) to ensure the reporter is updated and informed of current actions and next steps.
```

Any vulnerability information will be treated as confidential and will be not shared outside the security response process. Vulnerability information will not be publicly shared or disclosed without the consent of the reporter.

> # Public Disclosure

Reviewer: ```suggestion
# Public disclosure policy

```

> The public disclosure date of any security vulnerability will be negotiated between SIG-Security and the vulnerability reporter. The SIG aims to publicly disclose the issue as soon as possible, ideally once a mitigation as been identified. Disclosure time can range from almost immediately, for publicly known issues, to a few weeks depending on the complexity of issue and coordination required.

Reviewer 2: ideally once a mitigation `has` been identified

Author: Fixed.
